### PR TITLE
feat: add dart validate analysis fix commands

### PR DESCRIPTION
**Pull request by bestdan**
## Summary

Adds comprehensive Dart validation and fix commands for mono-repo support, enabling targeted package-based validation and linting. Introduces `dart fix` command with dry-run/apply modes, and a suite of `dart validate` commands (format, analysis, DCM, freezed) that intelligently work with both PACKAGE_INDEX-based mono-repos and standalone packages.

## Changes

- **New `dart fix` command**: Runs `dart fix` on affected packages or individual files, with dry-run (default) and apply modes
- **New `dart validate format` command**: Validates Dart formatting using `dart format --set-exit-if-changed` on affected packages
- **New `dart validate analysis` command**: Runs `dart analyze --fatal-infos` on affected packages
- **New `dart validate dcm` command**: Runs DCM analysis on affected packages (gracefully skips if not installed)
- **New `dart validate freezed` command**: Validates freezed-generated files are up to date by running build_runner
- **New `dart validate all` command**: Runs all validation checks in sequence with optional skip flags
- **PACKAGE_INDEX support**: All validate commands support optional PACKAGE_INDEX file for efficient package lookup in large mono-repos, with automatic fallback to pubspec.yaml walking
- **Dart utilities enhancement**: Added `findAffectedPackages()`, `readPackageIndex()`, and `readPackageName()` functions to `src/utils/dart.ts`
- **Comprehensive test coverage**: Added test files for all new commands validating error handling, edge cases, and mono-repo scenarios
- **Updated README**: Documented new commands, PACKAGE_INDEX format, and mono-repo usage patterns

## Testing

All new commands include comprehensive unit tests covering:
- Command installation checks and graceful fallback behavior
- Package discovery with PACKAGE_INDEX and pubspec.yaml fallback
- Success and failure scenarios for each validation type
- File filtering and generated file exclusion
- Error handling and exit code validation

### CODE REVIEW

File: src/commands/dart-validate-freezed.ts [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+  if (verbose) {
+    console.error('✓ All freezed files are up to date');
+  }
+  process.exit(0);
+}
+
+/**
+ * Finds the package root for a given file by walking up the directory tree
+ */
+function findPackageRootForFile(filePath: string): string | null {
```
[the remark this answers is not in the document]
Comment (bestdan) [reply src/commands/dart-validate-freezed.ts]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/bestdan/tsu/pull/28#discussion_r2484973026)

File: src/utils/dart.ts [hunk 2]
```diff
@@ -295,3 +295,141 @@ export function findDownstreamDependencies(
...
+
+/**
+ * Reads and parses PACKAGE_INDEX file from the workspace root.
+ * @param workspaceRoot - Root directory of the workspace (defaults to current directory)
+ * @returns Array of package entries, or null if file doesn't exist or is invalid
+ */
+export function readPackageIndex(
+  workspaceRoot: string = process.cwd()
+): PackageIndexEntry[] | null {
+  const packageIndexPath = resolve(workspaceRoot, 'PACKAGE_INDEX');
```
Comment (bestdan) [inline src/utils/dart.ts]: if this is a file path we're creating, let's use `TSU_PACKAGE_INDEX` for clarity

File: .gitignore [hunk 3]
```diff
@@ -1,5 +1,6 @@
 # Dependencies
 node_modules/
+package-lock.json
```
Comment (bestdan) [inline .gitignore]: why do we want to ignore this?

**Review by bestdan — COMMENTED** [review]
@copilot  please make updates based on the suggestions belo
